JASP-Engine/JASP/R/bainttestbayesianindependentsamples.R

```diff
@@ -24,335 +24,357 @@ BainTTestBayesianIndependentSamples <- function(jaspResults, dataset, options, .
+	descriptivesTable$addColumnInfo(name="mean",                 title = "Mean", type="number")
+	descriptivesTable$addColumnInfo(name="sd",                   title = "SD", type="number")
+	descriptivesTable$addColumnInfo(name="se",                   title = "SE", type="number")
 
-		interval <- 100 * options[["descriptivesPlotsCredibleInterval"]]
-		overTitle <- paste0(interval, "% Credible Interval")
-		descriptivesTable$addColumnInfo(name="lowerCI",              title = "Lower", type="number", overtitle = overTitle)
-	  descriptivesTable$addColumnInfo(name="upperCI",              title = "Upper", type="number", overtitle = overTitle)
-		
-		jaspResults[["descriptivesTable"]] <- descriptivesTable
-
-		if (!ready)
-			return()
-
-		levels <- base::levels(dataset[[ .v(options$groupingVariable) ]])
-		if (length(levels) != 2) {
-			g1 <- "1"
-			g2 <- "2"
-		} else {
-			g1 <- levels[1]
-			g2 <- levels[2]
-		}
+	overTitle <- paste0(100 * options[["credibleInterval"]], "% Credible Interval")
+	descriptivesTable$addColumnInfo(name="lowerCI",              title = "Lower", type="number", overtitle = overTitle)
+	descriptivesTable$addColumnInfo(name="upperCI",              title = "Upper", type="number", overtitle = overTitle)
+	
+	bainContainer[["descriptivesTable"]] <- descriptivesTable
+
+	if (!ready || bainContainer$getError())
+		return()
+
+	descriptivesTable$setExpectedSize(length(options[["variables"]]) * 2)
+
+	levels <- base::levels(dataset[[ .v(options[["groupingVariable"]]) ]])
+	if (length(levels) != 2) {
+		g1 <- "1"
+		g2 <- "2"
+	} else {
+		g1 <- levels[1]
+		g2 <- levels[2]
+	}
 
 	for (variable in options[["variables"]]) {
-		for (i in 1:2) {
-
-	  	level <- levels[i]
-	  	variableData <- dataset[[.v(variable)]]
-			groupingData <- dataset[[.v(options$groupingVariable)]]
-			groupData <- variableData[groupingData == level]
-			groupDataOm <- na.omit(groupData)
-
-			if (class(groupDataOm) != "factor") { # TODO: Fix this...
-				posteriorSummary <- .posteriorSummaryGroupMean(variable=groupDataOm, descriptivesPlotsCredibleInterval=options$descriptivesPlotsCredibleInterval)
-				ciLower <- round(posteriorSummary$ciLower,3)
-				ciUpper <- round(posteriorSummary$ciUpper,3)
-				n <- length(groupDataOm)
-		  		mean <- mean(groupDataOm)
-		  		std <- sd(groupDataOm)
-		  		sem <- sd(groupDataOm) / sqrt(length(groupDataOm))
-				if (i == 1)
-					row <- data.frame(v = variable, group = level, N = n, mean = mean, sd = std, se = sem, lowerCI = ciLower, upperCI = ciUpper)
-				if (i == 2)
-					row <- data.frame(v = "", group = level, N = n, mean = mean, sd = std, se = sem, lowerCI = ciLower, upperCI = ciUpper)
-
-				} else {
-						n <- length(groupDataOm)
-						row <- data.frame(v = variable, group = "", N = n, mean = "", sd = ".", se = "", lowerCI = "", upperCI = "")
-				}
-				descriptivesTable$addRows(row)
-			}	
-  	}
+		
+		bainAnalysis <- .bainTwoSampleState(variable, options, dataset, bainContainer)
```

Comment:
I'm not sure what you mean? This is a try-error right?

Reply:
I see what you mean now!

Reply:
In `.bainTwoSampleState()` you put a `try()` around the main calculation. So that calculation can obviously fail (and you handle this possible failure well in `bainIndependentSamplesResultsTable`), however in this function you don't handle this possible failure. If you call `bainSummary <- summary(bainAnalysis, ci = options[["credibleInterval"]])` on an object of class `try-error` you'll get bad things.

Reply:
Implemented